languages/javascript/stocktwits-api-client.js

```diff
@@ -0,0 +1,458 @@
+'use strict';
+
+const readline = require('readline')
+const axios = require('axios');
+
+const commandSchema = `
+Available Command Schema
+
+<action> <resource> <identifier> <value/term>
+
+get user <username or id>
+get symbol <symbol or id>
+get message <message id>
+
+list blocks <username or id>
+list mutes <username or id>
+list followers <username or id>
+list follows <username or id>
+list likes <username or id>
+list watchlist <username or id>
+
+list messages user <username or id>
+list messages symbol <symbol or id>
+list messages topic <topic or id>
+`;
+
+String.prototype.format = function() {
+
+  let currentString = this;
+
+  for (const argumentIndex in arguments) {
+    let regex = new RegExp(`\\{${argumentIndex}\\}`, 'g');
+    currentString = currentString.replace(regex, arguments[argumentIndex]);
+  }
+
+  return currentString;
+
+}
+
+function stringify(obj) {
+  return JSON.stringify(obj, null, 2);
+}
+
+const config = {
+  debug: false,
+
+  targetAPIEndpoint: 'https://api.stocktwits.com/api/2/',
```

Comment:
I added this but never used it - you'll see hardcoding/duplication across all the methods that touch the API endpoint. In the real world, we'd ideally want to centralize/encapsulate endpoints like this, then use proper `url.combine(...)` utils to build any endpoints we touch. For this simple example though, I just left things as is.